packages/dart_firebase_admin/lib/src/app/credential.dart

```diff
@@ -25,6 +25,22 @@ class Credential {
     return Credential._(serviceAccountCredentials);
   }
 
+  /// Log in to firebase from a service account file parameters.
+  factory Credential.fromServiceAccountParams({
+    required String clientId,
+    required String privateKey,
+    required String clientEmail,
+  }) {
+    final serviceAccountCredentials = auth.ServiceAccountCredentials.fromJson({
+      'type': 'service_account',
+      'private_key': privateKey,
+      'client_email': clientEmail,
+      'client_id': clientId,
+    });
```

Comment:
No need for fromJSON:

```suggestion
    final serviceAccountCredentials = auth.ServiceAccountCredentials(
      email,
      clientId,
      privateKey,
    );
```